mlrun/utils/notifications/notification_pusher.py

```diff
@@ -328,14 +329,22 @@ async def _async_push():
                     tasks.append(
                         notification.push(message, severity, runs, custom_html)
                     )
-
             # return exceptions to "best-effort" fire all notifications
             await asyncio.gather(*tasks, return_exceptions=True)
 
         # first push async notifications
         main_event_loop = asyncio.get_event_loop()
         if main_event_loop.is_running():
-            asyncio.run_coroutine_threadsafe(_async_push(), main_event_loop)
+            # Check if the code is running inside a Jupyter Notebook
+            # if so, we need to create a new thread and run a separate event loop on that thread,
+            # and use it instead of the main_event_loop
+            # This is necessary because Jupyter Notebook has its own event loop,
+            # but it runs in the main thread. As long as a cell is running,
+            # the event loop will not execute properly
+            if mlrun.utils.helpers.is_running_in_jupyter_notebook():
+                _run_coroutine_in_jupyter_notebook(coroutine_method=_async_push)
+            else:
+                asyncio.run_coroutine_threadsafe(_async_push(), main_event_loop)
         else:
             main_event_loop.run_until_complete(_async_push())
```

Comment:
```suggestion
        if not main_event_loop.is_running():
            # Running in local python process
            main_event_loop.run_until_complete(_async_push())
        elif mlrun.utils.helpers.is_running_in_jupyter_notebook():
            # Running in Jupyter notebook.
            # In this case, we need to create a new thread, run a separate event loop in 
            # that thread, and use it instead of the main_event_loop.
            # This is necessary because Jupyter Notebook has its own event loop,
            # but it runs in the main thread. As long as a cell is running,
            # the event loop will not execute properly
            _run_coroutine_in_jupyter_notebook(coroutine_method=_async_push)
        else:
            # Running in mlrun api, we are in a separate thread from the one in which 
            # the main event loop, so we can just send the notifications to that loop
            asyncio.run_coroutine_threadsafe(_async_push(), main_event_loop)
```